src/lib.rs

```diff
@@ -558,11 +558,15 @@ fn encode_block(fi: &FrameInvariants, fs: &mut FrameState, cw: &mut ContextWrite
         cw.bc.reset_skip_context(bo, bsize, xdec, ydec);
     }
 
-    let tx_type = if tx_size > TxSize::TX_32X32 {
-        TxType::DCT_DCT
-    } else {
+    // Luma plane transform type decision
+    let square_tx_size = TXSIZE_SQR_MAP[tx_size as usize];
+    let tx_set_type = get_ext_tx_set_type(square_tx_size, is_inter, fi.use_reduced_tx_set);
```

Comment:
get_ext_tx_set_type already does the tx squaring, doesn't it?

Reply:
Indeed, thanks for pointing that out.